graph/src/firehose/endpoints.rs

```diff
@@ -267,10 +264,12 @@ impl FirehoseEndpoints {
         self.0.len()
     }
 
+    // selects the FirehoseEndpoint with the lest amount of references, which will help with spliting
+    // the load naively across the entire list.
     pub fn random(&self) -> Option<&Arc<FirehoseEndpoint>> {
```

Comment:
One aspect that concerns me is a silent hang if we ever hit the 100 stream limit per connection limit. Perhaps a simple way to make the error explicit would be to add a
```
const SUBGRAPHS_PER_CONN: usize = 100;
```
And return an error if the ref count reaches this number. Or even better auto-scale based on this.

Reply:
will address this on a separate PR as this could benefit from some metrics in order to detect a stall